staging/src/k8s.io/apiserver/pkg/admission/plugin/policy/validating/typechecking.go

```diff
@@ -29,17 +28,13 @@ import (
 	"k8s.io/apimachinery/pkg/api/meta"
 	"k8s.io/apimachinery/pkg/runtime/schema"
 	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
-	"k8s.io/apimachinery/pkg/util/sets"
 	"k8s.io/apimachinery/pkg/util/validation/field"
 	"k8s.io/apimachinery/pkg/util/version"
 	plugincel "k8s.io/apiserver/pkg/admission/plugin/cel"
 	apiservercel "k8s.io/apiserver/pkg/cel"
-	"k8s.io/apiserver/pkg/cel/common"
 	"k8s.io/apiserver/pkg/cel/environment"
 	"k8s.io/apiserver/pkg/cel/library"
-	"k8s.io/apiserver/pkg/cel/openapi"
 	"k8s.io/apiserver/pkg/cel/openapi/resolver"
-	"k8s.io/klog/v2"
 )
 
 const maxTypesToCheck = 10
```

Comment:
Do we have test for limitation handling(11 warnings originally and one fixed, will it result in 10 warning or 9?)

Reply:
Nope. However let's limit the scope of this PR to only the "rechecking" part. Let me quickly send another PR for it.